src/main/java/de/teamlapen/vampirism/entity/hunter/AdvancedHunterEntity.java

```diff
@@ -334,11 +334,13 @@ private void applyCustomisationItems(Supporter supporter) {
         this.setItemSlot(EquipmentSlot.MAINHAND, equipment.getMainHand());
         this.setItemSlot(EquipmentSlot.OFFHAND, equipment.getOffHand());
 
-        HatType hat = Optional.ofNullable(appearance.get("hat")).map(HatType::get).orElseGet(() -> {
+        ResourceLocation headwearId = ResourceLocation.parse(appearance.get("headwear"));
```

Comment:
the headwear might not be set or might be invalid so we need to first check for null and catch the parse exception that might be thrown.

If the parsing fails or the item is not present, we should not assign an item and log a warning.
If none is set we should assign a random one.